nni/retiarii/execution/cgo_engine.py

```diff
@@ -143,8 +148,14 @@ def trial_execute_graph(cls) -> None:
 
 class AssemblePolicy:
     @staticmethod
-    def group(logical_plan):
+    def group(logical_plan, n_model_per_graph):
+        #TODO: Packing multiple model in one GPU
+        # Currently, we only support one model per GPU
+        all_grouped_models = []
         group_model = {}
         for idx, m in enumerate(logical_plan.models):
             group_model[m] = PhysicalDevice('server', f'cuda:{idx}')
```

Comment:
there is a bug here, users can submit many models, then logical plan merge them together. You enumerate them here and assign gpu with model's `idx`, it is possible that `idx` is out of range

Reply:
Fixed. I add "available_devices" in CGOExecutionEngine. I also change CGO in the environment variable to CGO_DEVICES to allow users to set available devices.